### ISSUE TITLE
Warnings for duplicate labels when using arguments in \newenvironment

### ISSUE THREAD

**Issue by rchoudhary** [James-Yu/LaTeX-Workshop#3395]
### Disable all the other extensions except for LaTeX Workshop, restart VS Code, and check that you still see this issue. [Required]

You still see this issue?: Yes

### Make sure to visit the [wiki FAQ](https://github.com/James-Yu/LaTeX-Workshop/wiki/FAQ) before filling an issue.

You visited the wiki?: Yes

### If your issue is with compiling a document (not having to do with finding the root file of a project), check first that you can compile manually.

You can compile a TeX document manually?: Yes

## Describe the bug [Required]

The extension seems to flag an argument to `\newenvironment` as a "duplicate label". Here is a minimal example that reproduces the problem:

```
\documentclass{article}

\newenvironment{env1}{%
    \label{#2}
}

\newenvironment{env2}{%
    \label{#2}
}

\title{Document Title}
\author{Document Author}
\date{}

\begin{document}

\maketitle

\end{document}
```

A similar issue was brought up before: #2785; and a PR merged to fix that issue: #2786. However, those only addressed the warnings occurring for `\newcommand`. I imagine the fix will be same to that PR: just need to check if the duplicate label is within a `\newenvionment` directive.

## Desktop [Required]

Please write exact version numbers. Please don't write `latest` instead of exact numbers.

 - OS: Mac OS 12.4
 - VS Code version: 1.69.2
 - Extension version: 8.27.2
 - TeX distribution version: mactex-no-gui 2022.0321 (homebrew package)

### LaTeX Workshop Output [Required]

```
[22:57:03] Initializing LaTeX Workshop.
[22:57:03] Extension root: /Users/rutvik/.vscode/extensions/james-yu.latex-workshop-8.27.2
[22:57:03] $PATH: /Applications/Julia-1.7.app/Contents/Resources/julia/bin:/opt/homebrew/opt/llvm/bin:/opt/homebrew/opt/m4/bin:/Users/rutvik/spack/bin:/opt/homebrew/opt/openjdk/bin:/opt/homebrew/opt/grep/libexec/gnubin:/opt/homebrew/bin:/opt/homebrew/sbin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/Library/TeX/texbin
[22:57:03] $SHELL: /bin/zsh
[22:57:03] $LANG: undefined
[22:57:03] $LC_ALL: undefined
[22:57:03] process.platform: darwin
[22:57:03] process.arch: arm64
[22:57:03] vscode.env.appName: Visual Studio Code
[22:57:03] vscode.env.remoteName: undefined
[22:57:03] vscode.env.uiKind: 1
[22:57:03] Workspace for configuration: file:///Users/rutvik/Desktop/tmp
[22:57:03] editor.acceptSuggestionOnEnter: "on"
[22:57:03] latex-workshop.bind.enter.key: true
[22:57:03] latex-workshop.docker.enabled: false
[22:57:03] latex-workshop.docker.image.latex: ""
[22:57:03] latex-workshop.hover.preview.mathjax.extensions: []
[22:57:03] latex-workshop.intellisense.package.enabled: true
[22:57:03] latex-workshop.intellisense.update.aggressive.enabled: false
[22:57:03] latex-workshop.intellisense.update.delay: 1000
[22:57:03] latex-workshop.latex.autoBuild.run: "onFileChange"
[22:57:03] latex-workshop.latex.build.forceRecipeUsage: true
[22:57:03] latex-workshop.latex.outDir: "%DIR%"
[22:57:03] latex-workshop.latex.recipes: [
 {
  "name": "latexmk 🔃",
  "tools": [
   "latexmk"
  ]
 },
 {
  "name": "latexmk (latexmkrc)",
  "tools": [
   "latexmk_rconly"
  ]
 },
 {
  "name": "latexmk (lualatex)",
  "tools": [
   "lualatexmk"
  ]
 },
 {
  "name": "latexmk (xelatex)",
  "tools": [
   "xelatexmk"
  ]
 },
 {
  "name": "pdflatex ➞ bibtex ➞ pdflatex × 2",
  "tools": [
   "pdflatex",
   "bibtex",
   "pdflatex",
   "pdflatex"
  ]
 },
 {
  "name": "Compile Rnw files",
  "tools": [
   "rnw2tex",
   "latexmk"
  ]
 },
 {
  "name": "Compile Jnw files",
  "tools": [
   "jnw2tex",
   "latexmk"
  ]
 },
 {
  "name": "tectonic",
  "tools": [
   "tectonic"
  ]
 }
]
[22:57:03] latex-workshop.latex.tools: [
 {
  "name": "latexmk",
  "command": "latexmk",
  "args": [
   "-xelatex",
   "-synctex=1",
   "-interaction=nonstopmode",
   "-file-line-error",
   "-shell-escape",
   "-halt-on-error",
   "%DOC%"
  ],
  "env": {
   "PATH": "/usr/bin:/Library/TeX/texbin:/opt/homebrew/bin"
  }
 }
]
[22:57:03] latex-workshop.viewer.pdf.internal.keyboardEvent: "auto"
[22:57:03] Creating a new file watcher.
[22:57:03] watcherOptions: {"useFsEvents":false,"usePolling":false,"interval":300,"binaryInterval":1000,"awaitWriteFinish":{"stabilityThreshold":250}}
[22:57:03] Creating PDF file watcher.
[22:57:03] watcherOptions: {"useFsEvents":false,"usePolling":false,"interval":300,"binaryInterval":1000,"awaitWriteFinish":{}}
[22:57:03] Creating Bib file watcher.
[22:57:03] watcherOptions: {"useFsEvents":false,"usePolling":false,"interval":300,"binaryInterval":1000,"awaitWriteFinish":{"stabilityThreshold":250}}
[22:57:03] Set $LATEXWORKSHOP_DOCKER_LATEX: ""
[22:57:03] [Server] Creating LaTeX Workshop http and websocket server.
[22:57:03] LaTeX Workshop initialized.
[22:57:03] Trigger characters for intellisense of LaTeX documents: ["\\",",","{"]
[22:57:03] Bibtex format config: {"tab":"  ","case":"lowercase","left":"{","right":"}","trailingComma":false,"sort":["key"],"alignOnEqual":true,"sortFields":false,"fieldsOrder":[],"firstEntries":["string","xdata"]}
[22:57:03] Current workspace folders: ["file:///Users/rutvik/Desktop/tmp"]
[22:57:03] Found root file from active editor: /Users/rutvik/Desktop/tmp/main.tex
[22:57:03] Root file changed: from undefined to /Users/rutvik/Desktop/tmp/main.tex
[22:57:03] Start to find all dependencies.
[22:57:03] Root file languageId: latex
[22:57:03] [Server] Server successfully started: {"address":"127.0.0.1","family":"IPv4","port":52323}
[22:57:03] Reset file watcher.
[22:57:03] Parsing a file and its subfiles: /Users/rutvik/Desktop/tmp/main.tex
[22:57:03] Parse fls file.
[22:57:03] Fls file found: /Users/rutvik/Desktop/tmp/main.fls
[22:57:03] Parse aux file: /Users/rutvik/Desktop/tmp/main.aux
[22:57:03] Added to file watcher: /Users/rutvik/Desktop/tmp/main.tex
[22:57:03] Snippet data loaded.
[22:57:03] Checking for duplicate labels: /Users/rutvik/Desktop/tmp/main.tex.
[22:57:03] [Server] valdOrigin is http://127.0.0.1:52323
[22:57:05] Manager.fileWatcher.getWatched: {"/Users/rutvik/Desktop/tmp":["main.tex"]}
[22:57:05] Manager.filesWatched: ["/Users/rutvik/Desktop/tmp/main.tex"]
[22:57:05] BibWatcher.bibWatcher.getWatched: {}
[22:57:05] BibWatcher.bibsWatched: []
[22:57:05] PdfWatcher.pdfWatcher.getWatched: {}
[22:57:05] PdfWatcher.pdfsWatched: []
[22:57:05] PdfWatcher.watchedPdfVirtualUris: []
[22:57:05] PdfWatcher.ignoredPdfUris: []
[22:57:08] onDidSaveTextDocument triggered: file:///Users/rutvik/Desktop/tmp/main.tex
[22:57:08] File watcher - file changed: /Users/rutvik/Desktop/tmp/main.tex
[22:57:08] Parsing a file and its subfiles: /Users/rutvik/Desktop/tmp/main.tex
[22:57:09] Checking for duplicate labels: /Users/rutvik/Desktop/tmp/main.tex.
[22:57:09] Auto build started detecting the change of a file: /Users/rutvik/Desktop/tmp/main.tex
[22:57:09] BUILD command invoked.
[22:57:09] The document of the active editor: file:///Users/rutvik/Desktop/tmp/main.tex
[22:57:09] The languageId of the document: latex
[22:57:09] Building root file: /Users/rutvik/Desktop/tmp/main.tex
[22:57:09] Build root file /Users/rutvik/Desktop/tmp/main.tex
[22:57:09] outDir: /Users/rutvik/Desktop/tmp
[22:57:09] Recipe step 1: latexmk
[22:57:09] Recipe step 1 args: ["-xelatex","-synctex=1","-interaction=nonstopmode","-file-line-error","-shell-escape","-halt-on-error","/Users/rutvik/Desktop/tmp/main"]
[22:57:09] Recipe step env: {"PATH":"/usr/bin:/Library/TeX/texbin:/opt/homebrew/bin"}
[22:57:09] cwd: /Users/rutvik/Desktop/tmp
[22:57:09] LaTeX build process spawned. PID: 32421.
[22:57:09] Recipe of length 1 finished. PID: 32421.
[22:57:09] Successfully built /Users/rutvik/Desktop/tmp/main.tex.
```

### Are you using VSCodium?

No

### Are you using the Snap or Flatpack versions of VS Code?

No

### Are you using LaTeX Workshop with VS Code Remote?

No
